citation-backoffice/src/logic/schema.js

```diff
@@ -0,0 +1,82 @@
+import {createAction, createReducer} from 'redux-act';
+import {query} from './graphql-client';
+
+export const loadSchemaSuccess = createAction('load schema success');
+export const loadSchemaFieldsSuccess = createAction('load schema fieds success');
+
+export async function queryExistingTypes() {
+	return query(`{
+			__schema {
+				types {
+					name, kind
+				}
+			}
+		}`);
+}
+
+export async function queryCustomTypes(type) {
+	const result = {};
+	const types = await query(`{
+		__type(name: "${type}") {
+				name
+				fields {
+					name
+					type {
+						name
+						kind
+					}
+				}
+			}
+		}`);
+	types.data.__type.fields.forEach(field => {
+		result[field.name] = {};
+		result[field.name].typeName = field.type.name ? field.type.name : (field.name === 'children' ? type : '*');
+		result[field.name].kind = field.type.kind;
+	});
+	return result;
+}
+
+async function askTypes() {
+	return queryExistingTypes().then(response => {
```

Comment:
Should not be a `then` in an `async` function.

There is some async tricks to fix in this file